### DISCUSSION TITLE
how to make a simple check

### DISCUSSION THREAD

**Discussion by MAngel666** [mpounsett/nagiosplugin#51] [Q&A]
Hi all,

how can I make a simple check with something like this:

if a == 1 return Ok
if a == 2 return Critical

can you make a such example?

I've already tried this:
```
import nagiosplugin

class CheckA(nagiosplugin.Resource):
    def probe(self):
        a = 1  
        if a == 1:
            return nagiosplugin.Result(nagiosplugin.Ok, 'a is 1')
        elif a == 2:
            return nagiosplugin.Result(nagiosplugin.Critical, 'a is 2')
        else:
            return nagiosplugin.Result(nagiosplugin.Unknown, 'a is unknown')

@nagiosplugin.guarded
def main():
    check = nagiosplugin.Check(CheckA())
    check.main()

if __name__ == '__main__':
    main()
```

I get this error: AttributeError: 'Ok' object has no attribute 'context'
What I'm doing wrong?

TiA

**Comment by mpounsett** [comment]
You have only a Resource, and you're trying to directly return a Result.  You're missing the Metric and its Context for evaluating what condition it's in.

Please have a look at the [First Steps](https://nagiosplugin.readthedocs.io/en/stable/tutorial/index.html) section of the manual.  The examples there are also distributed with the source code, and you can use those to play around with.  Extending the Hello World example (`check_world.py`) is probably a good place to start.

**Comment by MAngel666** [comment]
hmmm I do not understand how can I make the different states here:
```
def probe(self):
        return [nagiosplugin.Metric('world', True, context='null')]
```
I dont have thresholds etc, only if foo then ok, if bar then critical etc...

an example would be really cool...
Sorry, I'm a newbe in the python OOP world...

**Comment by MAngel666** [comment]
I'm I on the right way? :)

```
import nagiosplugin

class Check_A(nagiosplugin.Resource):
    def probe(self):
        a = 1
        return [nagiosplugin.Metric('a', a, context='a')]

class Check_A(nagiosplugin.Context):
    def evaluate(self):
        if a == 1:
            return nagiosplugin.Result(nagiosplugin.Ok, 'a is 1', context='a')
        if a == 2:
            return nagiosplugin.Result(nagiosplugin.Critical, 'a is 2', context='a')

@nagiosplugin.guarded
def main():
    check = nagiosplugin.Check(
            Check_A(),
            nagiosplugin.Context('a')
    )

    check.main()

if __name__ == '__main__':
    main()
```
